src/slice.cr

```diff
@@ -493,11 +495,11 @@ struct Slice(T)
     end
   end
 
-  private def to_hex(c)
+  private def to_hex(c : Int) : UInt8
     ((c < 10 ? 48_u8 : 87_u8) + c)
   end
 
-  def bytesize
+  def bytesize : Int32
     sizeof(T) * size
```

Comment:
Whoa, this may be a candidate for being upscaled to Int64.

Reply:
or maybe this could just be `Int`

Reply:
The problem is with the implementation, not the type. It will be wrong if sizeof(T) * size > Int32::MAX.

Reply:
That makes sense but I would want to pull that into another PR